doc/authentication.md

```diff
@@ -0,0 +1,162 @@
+
+Any generated key should be at least 4096 bit RSA, preferably EC384.
+When using signatures, only SHA-2 signatures should be trusted.
+
+Since we control both client and server, there is no reason to support
+any backward compatibility to broken protocol or ciphers.
+
+(authentication-trusted-clients)=
+### Trusted TLS clients
+
+You can obtain the list of TLS certificates trusted by a LXD server with `lxc config trust list`.
+
+Trusted clients can be added in either of the following ways:
+
+- {ref}`authentication-add-certs`
+- {ref}`authentication-trust-pw`
+
+To revoke trust to a client, remove its certificate from the server with `lxc config trust remove FINGERPRINT`.
+
+It's possible to restrict a TLS client to one or multiple projects.
+In this case, the client will also be prevented from performing global configuration changes or altering the configuration (limits, restrictions) of the projects it's allowed access to.
+
+To restrict access, use `lxc config trust edit FINGERPRINT`.
+Set the `restricted` key to `true` and specify a list of projects to restrict the client to.
+If the list of projects is empty, the client will not be allowed access to any of them.
+
+(authentication-add-certs)=
+#### Adding trusted certificates to the server
+
+The preferred way to add trusted clients is to directly add their certificates to the trust store on the server.
+To do so, copy the client certificate to the server and register it using `lxc config trust add <file>`.
+
+(authentication-trust-pw)=
+#### Adding client certificates using a trust password
+
+To allow establishing a new trust relationship from the client side, you must set a trust password (`core.trust_password`, see {doc}`server`) for the server. Clients can then add their own certificate to the server's trust store by providing the trust password.
+
+The workflow to authenticate with the server is similar to that of SSH, where an initial connection to an unknown server triggers a prompt:
+
+1. When the user adds a server with `lxc remote add`, the server is contacted over HTTPS, its certificate is downloaded and the fingerprint is shown to the user.
+1. The user is asked to confirm that this is indeed the server's fingerprint, which they can manually check by connecting to the server or by asking someone with access to the server to run the info command and compare the fingerprints.
+1. The user enters the trust password for the server.
+1. If the trust password matches, the client certificate is added to the server's trust store. The client can now connect to the server without having to provide any additional credentials.
+
+A remote add operation should therefore be implemented like this:
+
+1. Call GET /1.0.
+1. Ask the user to confirm the fingerprint.
+1. Look at the dict received from the server. If "auth" is "untrusted":
+
+   1. Ask the user for the server's password.
+   1. Do a `POST` to `/1.0/certificates`.
+   1. Call `/1.0` again to check that the client is now trusted.
```

Comment:
In a future update of this doc, we probably ought to get rid of this. We haven't really been providing such API workflow/examples throughout the doc so having this one here feels weird. If we find the need, we could always add something like this back as a howto.